### PR TITLE
Added segmented button,segmented button style, and example

### PR DESCRIPTION
**Pull request by bq-wrongway**
This is just a preliminary pull request, i am looking forward to your feedback in order to change, addend and improve this pull merge request.

Thank you!

### CODE REVIEW

File: examples/segmented_button/resources/dl.svg
Comment (genusistimelord) [inline examples/segmented_button/resources/dl.svg:1]: these svgs do not belong here?

File: examples/segmented_button/resources/file_add.png
Comment (genusistimelord) [inline examples/segmented_button/resources/file_add.png:1]: the image does not belong here. as i did not see it being used in the example

File: examples/segmented_button/resources/ico.svg
Comment (genusistimelord) [inline examples/segmented_button/resources/ico.svg:1]: the svg does not belong here. as i did not see it being used in the example

File: examples/segmented_button/resources/light.svg
Comment (genusistimelord) [inline examples/segmented_button/resources/light.svg:1]: the svg does not belong here. as i did not see it being used in the example

**Review by genusistimelord — CHANGES_REQUESTED** [review]
remove the image and svgs then this can be merged.

**Comment by bq-wrongway** [comment]
You are right, i forgot about those.
I have deleted the whole folder since i was just using that to test and preview.

**Review by genusistimelord — APPROVED** [review]
looks good once the checks have passed I can merge this in.

**Comment by genusistimelord** [comment]
looks like there are some formatting and clippy issues you need to fix.

**Comment by bq-wrongway** [comment]
Hey, i have fixed both linting and formating issues, and on my machine they are passing all the checks at the moment!
Thank you for your patience

**Review by genusistimelord — APPROVED** [review]

**Pull request merged**